### ISSUE TITLE
Error 400 returned for some clients connecting via IPv6

### ISSUE THREAD

**Issue by TuxPowered42** [jkroepke/openvpn-auth-oauth2#125]
### Current Behavior

I've configured openvpn-auth-oauth2 to use Azure as OAuth provider with the following config:

```
CONFIG_OAUTH2_ISSUER=https://login.microsoftonline.com/…/v2.0
CONFIG_OAUTH2_CLIENT_ID=…
CONFIG_OAUTH2_CLIENT_SECRET=…

CONFIG_OPENVPN_ADDR=unix:///run/openvpn/server-udp-1194.sock
CONFIG_OPENVPN_PASSWORD=… # The same password as in /etc/openvpn/password.txt

CONFIG_HTTP_LISTEN=:9000
CONFIG_HTTP_SECRET=…
CONFIG_HTTP_BASEURL=https://vpn-test.$COMPANY.com/
CONFIG_HTTP_ENABLEPROXYHEADERS=true

CONFIG_LOG_LEVEL=debug
```

There's nginx to provide TLS:
```
server {
    listen 80 default_server;
    listen [::]:80 default_server;
    server_name vpn-test.$COMPANY.com;

    return 301 https://$host$request_uri;
}

server {
    listen 443 ssl default_server;
    listen [::]:443 ssl default_server;
    server_name vpn-test.$COMPANY.com;
    ssl_certificate /etc/ssl/private/star_$COMPANY_com_fullchain.pem;
    ssl_certificate_key /etc/ssl/private/star_$COMPANY_com.key;

    include /etc/nginx/include/well_known.conf;

    location / {
        proxy_pass http://localhost:9000;
        proxy_set_header X-Real-IP  $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header Host $host;
        proxy_set_header X-Real-Port $server_port;
        proxy_set_header X-Real-Scheme $scheme;
    }
}
```

In my tests I have tried a few client and platform combination and got different results:
- Viscosity on Windows. Pops up its own browser window where authentication works fine, I'm being redirected to Microsoft and so on.
- OpenVPN GUI on Windows. Opens the default web browser Edge with a page showing error 400. The same error is in logged by nginx.
- Viscosity on Mac. Pops ip its own window with blank page. Nginx log says it's a 400.

App's log:
```
2024-01-18T10:57:28+0000vpn-test.$COMPANY.local openvpn-auth-oauth2[423118]: time=2024-01-18T10:57:28.355Z level=INFO msg="new client connection" cid=0 kid=1 common_name="" reason=CONNECT username=""
2024-01-18T10:57:28+0000vpn-test.$COMPANY.local openvpn-auth-oauth2[423118]: time=2024-01-18T10:57:28.355Z level=INFO msg="start pending auth" cid=0 kid=1 common_name="" reason=CONNECT username=""
2024-01-18T10:57:28+0000vpn-test.$COMPANY.local openvpn-auth-oauth2[423118]: time=2024-01-18T10:57:28.355Z level=DEBUG msg="client-pending-auth 0 1 \"WEB_AUTH::https://vpn-test.$COMPANY.com/oauth2/start?state=A_LONG_RANDOM_STRING\" 180"
2024-01-18T10:57:28+0000vpn-test.$COMPANY.local openvpn-auth-oauth2[423118]: time=2024-01-18T10:57:28.415Z level=WARN msg="invalid state: decode: EOF"
2024-01-18T10:57:28+0000vpn-test.$COMPANY.local openvpn-auth-oauth2[423118]: time=2024-01-18T10:57:28.415Z level=DEBUG msg=A_LONG_RANDOM_STRING
```

The A_LONG_RANDOM_STRING is the same as in the URL opened in the pop-up browser window:
```
https://vpn-test.$COMPANY.com/oauth2/start?state=A_LONG_RANDOM_STRING
```

### Expected Behavior

URL https://vpn-test.$COMPANY.com/oauth2/start?state=A_LONG_RANDOM_STRING opening and redirecting me to Microsoft for all clients on all platforms.

### Environment

- openvpn-auth-oauth2 Version: 1.13.4
- OpenVPN Server Version: 2.6.3-1+deb12u2~bpo11+1
- Server OS: Debian 11.8
- OpenVPN Client (flavor, OS): OpenVPN GUI on Windows, Viscosity on Mac and Windows

**Comment by jkroepke** [comment]
can you share the logs without mask the A_LONG_RANDOM_STRING?

**Comment by TuxPowered42** [comment]
> common_name="" reason=CONNECT username=""

No stripping done, this is copied verbatim from the log.

**Comment by jkroepke** [comment]
Has the client "Viscosity on Windows" a common name or username?

**Comment by TuxPowered42** [comment]
> I improved the error message in v1.13.5, maybe it helps here

It seem it does, see:
```
2024-01-24T15:36:07+0000 vpn-test.$COMPANY.local openvpn-auth-oauth2[3094910]: time=2024-01-24T15:36:07.101Z level=WARN msg="invalid state: scan error \"4 1 \\x00 \\x00  \\x00 1706110402\\n\": unexpected newline"
```

**Comment by TuxPowered42** [comment]
> The logs without A_LONG_RANDOM_STRING mask would helpful here.

```
https://vpn-test.$COMPANY.com/oauth2/start?state=vCf4NKk1ls1VNkj_WZvBNUU1Kf0FJSHBlsMrT_JITNagGSdM1Y4
```

**Comment by jkroepke** [comment]
Could you please disable 

> CONFIG_HTTP_ENABLEPROXYHEADERS=true

**Comment by gruaig** [comment]
> can you share the logs without mask the A_LONG_RANDOM_STRING? its share to share, because the string is encrypted with the value of CONFIG_HTTP_SECRET
> 
> Did you also strip
> 
> > common_name="" reason=CONNECT username=""
> 
> or are the values empty?

These are also empty for OpenVPN Connect without stripping

**Comment by TuxPowered42** [comment]
The client connects from IPv6, not from Legacy IP and thus OpenVPN provides the env variable `untrusted_ip6` instead of `untrusted_ip`. See https://github.com/jkroepke/openvpn-auth-oauth2/pull/133 for bare minimum fix.

**Comment by jkroepke** [comment]
Resolved in #133
